zio-http/src/main/scala/zio/http/codec/QueryCodecs.scala

```diff
@@ -15,30 +15,23 @@
  */
 
 package zio.http.codec
-import zio.stacktracer.TracingImplicits.disableAutoTrace
+import zio.{Chunk, NonEmptyChunk}
 private[codec] trait QueryCodecs {
-  def query(name: String): QueryCodec[String] =
-    HttpCodec.Query(name, TextCodec.string)
+  @inline def queryAs[A](name: String)(implicit codec: TextCodec[A]): QueryCodec[A] =
+    HttpCodec.Query(name, TextChunkCodec.one(codec))
 
-  def queryBool(name: String): QueryCodec[Boolean] =
-    HttpCodec.Query(name, TextCodec.boolean)
+  def query(name: String): QueryCodec[String] = queryAs[String](name)
 
-  def queryInt(name: String): QueryCodec[Int] =
-    HttpCodec.Query(name, TextCodec.int)
+  def queryBool(name: String): QueryCodec[Boolean] = queryAs[Boolean](name)
 
-  def queryAs[A](name: String)(implicit codec: TextCodec[A]): QueryCodec[A] =
-    HttpCodec.Query(name, codec)
+  def queryInt(name: String): QueryCodec[Int] = queryAs[Int](name)
 
-  def paramStr(name: String): QueryCodec[String] =
-    HttpCodec.Query(name, TextCodec.string)
+  def queryOpt[I](name: String)(implicit codec: TextCodec[I]): QueryCodec[Option[I]] =
+    HttpCodec.Query(name, TextChunkCodec.optional(codec))
```

Comment:
I am pretty sure, that this is not necessary. We have already `HttpCodec#optional` that should cover this case.
Please try to replace `QueryCodec.queryOpt[X](y)` in the tests with `QueryCodec.query[X](y).optional`.
If it works, I think you can remove this constructor and `TextChunkCodec.optional`.
If not, please report back why it does not work.

Reply:
Yes, but it would be better to solve all cardinality cases in the same class and do not spread it over all code base.